src/iotjs_binding.cpp

```diff
@@ -116,13 +119,17 @@ JObject::~JObject() {
 
 
 JObject& JObject::Null() {
-  static JObject null(JVal::Null(), false);
-  return null;
+  if (null == NULL) {
+    null = new JObject(JVal::Null(), false);
```

Comment:
@chokobole I've taken a look at that commit. It's quite simple. It ensures static initialization in a single threaded context, but that's it, no thread safety. Didi it get accepted into Nuttx mainline?

As for threading in JS. Yes, JS is single threaded by default. (Although there are different approaches in different systems to allow some parallelization.) But... IoT.js is relying on libtuv, which _does_ use threading -- at least for the unix port, it surely uses pthreads. I'm not familiar enough with Nuttx to tell how things work there, but IoT.js should be OS agnostic, so we should consider all supported OSes (IMHO).

But I just bumped into an alternative potential solution, that can work on any OS and does not need static initializer guards. Roughly: `JObject` could have static `init` and `cleanup` functions that could instantiate/destroy `Null` and `Undefined` (on heap via `new` and `delete`), and `StartIoTjs` in iotjs.cpp could call them. This way no static initializer is needed, allocation surely happens only once, jerry api calls happen only after the api is available, and heap is deallocated when not needed. (This approach _does_ have the drawback, of course, that `JObject` as a class is only fully functional once its `init` method is called -- but AFAIK, that's all IoT.js-internal matter.)